File: Android - Native/discover_destination/app/src/main/java/com/example/discover_destination/repository/DestinationsRepository.kt
package com.example.discover_destination.repository

import com.example.discover_destination.DestinationsService
import com.example.discover_destination.domain.Destination
import javax.inject.Inject

interface DestinationsRepositoryInterface {
    fun add(destination: Destination)
    fun delete(city: String)
    fun modify(destination: Destination)
    suspend fun getAll(): List<Destination>
    fun getByCountry(country: String, user: String): List<Destination>
    fun getOne(city: String): Destination
}

class DestinationsRepository @Inject constructor(
    val service: DestinationsService
) : DestinationsRepositoryInterface {
    override fun add(destination: Destination){
        service.addDestination(destination)
    }

    override fun delete(city: String){
        service.deleteDestination(city)
    }

    override fun modify(destination: Destination) {
        service.modifyDestination(destination)
    }

    override suspend fun getAll(): List<Destination> {
        return service.getAllDestinations()
    }

    override fun getByCountry(country: String, user: String): List<Destination> {
        return service.getDestinationsByCountry(country, user)
    }

    override fun getOne(city: String): Destination {
        return service.getDestinationByCity(city)
    }
}

File: Flutter - Non-Native/discover_destination/lib/repository/destinations_repository.dart
import 'package:discover_destination/model/destination.dart';
import 'package:discover_destination/service/destinations_service.dart';

class DestinationsRepository {
  DestinationsService service;

  DestinationsRepository(this.service);

  void add(Destination destination) {
    service.addDestination(destination);
  }

  void delete(String city) {
    service.deleteDestination(city);
  }

  void modify(Destination destination) {
    service.modifyDestination(destination);
  }

  List<Destination> getAll(String user) {
    return service.getAllDestinations(user);
  }

  List<Destination> getByCountry(String country, String user) {
    return service.getDestinationsByCountry(country, user);
  }

  Destination getOne(String city) {
    return service.getDestinationByCity(city);
  }
}

